src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1191,7 +1191,11 @@ private function wrapServiceConditionals($value, $code)
 
         $conditions = array();
         foreach ($services as $service) {
-            $conditions[] = sprintf("\$this->has('%s')", $service);
+            if ($this->container->hasDefinition($service) && !$this->container->getDefinition($service)->isPublic()) {
+                $conditions[] = sprintf("isset(\$this->methodMap['%s'])", $service);
```

Comment:
in the "set" method, "methodMap" is not updated on override, whereas "privates" is,
shouldn't this check for "privates" ?